x-pack/plugins/observability_solution/observability_ai_assistant/server/service/knowledge_base_service/index.ts

```diff
@@ -92,137 +93,142 @@ export class KnowledgeBaseService {
   }
 
   setup = async () => {
-    const elserModelId = await this.dependencies.getModelId();
+    if (this.dependencies.enabled) {
+      const elserModelId = await this.dependencies.getModelId();
+
+      const retryOptions = { factor: 1, minTimeout: 10000, retries: 12 };
+      const getModelInfo = async () => {
+        return await this.dependencies.esClient.asInternalUser.ml.getTrainedModels({
+          model_id: elserModelId,
+          include: 'definition_status',
+        });
+      };
 
-    const retryOptions = { factor: 1, minTimeout: 10000, retries: 12 };
-    const getModelInfo = async () => {
-      return await this.dependencies.esClient.asInternalUser.ml.getTrainedModels({
-        model_id: elserModelId,
-        include: 'definition_status',
-      });
-    };
+      const isModelInstalledAndReady = async () => {
+        try {
+          const getResponse = await getModelInfo();
+          this.dependencies.logger.debug(
+            () =>
+              'Model definition status:\n' + JSON.stringify(getResponse.trained_model_configs[0])
+          );
+
+          return Boolean(getResponse.trained_model_configs[0]?.fully_defined);
+        } catch (error) {
+          if (isModelMissingOrUnavailableError(error)) {
+            return false;
+          } else {
+            throw error;
```

Comment:
nit: we can omit else there
```suggestion
          if (isModelMissingOrUnavailableError(error)) {
            return false;
          }
          
          throw error
```